File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& arr) {
        int MP=0,price=arr[0];
    for(int i=1;i<arr.size();i++){
        MP=max(MP,-price+arr[i]);
        price=min(price,arr[i]);
    }
    return MP;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxp=0,mp=prices[0];
        for(int i=1;i<prices.length;i++){
            maxp=Math.max(maxp,prices[i]-mp);
            mp=Math.min(mp,prices[i]);
        }
        return maxp;
    }
}